pkg/asset/installconfig/vsphere/validation.go

```diff
@@ -22,8 +22,8 @@ func Validate(ic *types.InstallConfig) error {
 	}
 
 	p := ic.Platform.VSphere
-	if err2 := validation.ValidatePlatform(p, field.NewPath("platform").Child("vsphere")); err2 != nil {
-		return err2.ToAggregate()
+	if err2 := validation.ValidatePlatform(p, field.NewPath("platform").Child("vsphere")).ToAggregate(); err2 != nil {
+		return err2
```

Comment:
I find it easier to look at this if we treat the result of `ValidatePlatform` as it is meant to be treated--as a collection of errors.
```suggestion
	if errs := validation.ValidatePlatform(p, field.NewPath("platform").Child("vsphere")); len(errs) > 0 {
		return errs.ToAggregate()
```